tests/phpunit/WC_Stripe_Webhook_Handler_Test.php

```diff
@@ -700,4 +701,91 @@ public function provide_test_process_webhook_charge_succeeded_skipped_for_synchr
 			'sepa_debit'     => [ WC_Stripe_Payment_Methods::SEPA_DEBIT ],
 		];
 	}
+
+	/**
+	 * Tests for `process_webhook_refund_updated`.
+	 *
+	 * @param string $notification_status The notification status.
+	 * @param bool   $email_triggered Whether an email should be triggered.
+	 * @param string $expected_note The expected order note.
+	 * @return void
+	 *
+	 * @dataProvider provide_test_process_webhook_refund_updated
+	 */
+	public function test_process_webhook_refund_updated( $notification_status, $email_triggered, $expected_note ) {
+		$refund_id = 'refund_123';
+		$charge_id = 'ch_123';
+
+		$order = WC_Helper_Order::create_order();
+		$order->set_payment_method( 'stripe' );
+		$order->set_transaction_id( $charge_id );
+		$order->save();
+
+		$order->update_meta_data( '_stripe_refund_id', $refund_id );
+		$order->save_meta_data();
+
+		$refund_order = WC_Helper_Order::create_order();
+		$refund_order->set_parent_id( $order->get_id() );
+		$refund_order->save();
+
+		$notification = (object) [
+			'data' => (object) [
+				'object' => (object) [
+					'id'             => $refund_id,
+					'charge'         => $charge_id,
+					'amount'         => 1000,
+					'failure_reason' => 'bank_account_rejected',
+					'status'         => $notification_status,
+				],
+			],
+		];
+
+		$this->mock_webhook_handler
+			->expects( $email_triggered ? $this->once() : $this->never() )
+			->method( 'send_failed_refund_emails' );
+
+		$this->mock_webhook_handler->process_webhook_refund_updated( $notification );
+
+		if ( ! empty( $expected_note ) ) {
+			// Check if the order note was added correctly.
+			$order = wc_get_order( $order->get_id() ); // Retrieve the order again to ensure we have the latest data.
+			if ( ! $order ) {
+				$this->fail( 'Order not found when testing the webhook refund updated method.' );
+			}
+
+			$notes = wc_get_order_notes(
+				[
+					'order_id' => $order->get_id(),
+					'limit'    => 1,
+				]
+			);
+			$this->assertCount( 1, $notes );
+			$this->assertSame( $expected_note, $notes[0]->content );
+		}
```

Comment:
Should we also verify that no notes are added when `$expected_note` is empty?

Also, I am not sure why we are (re)fetching the order, when all we need to look at are the notes for the order?
```suggestion
		$notes = wc_get_order_notes(
			[
				'order_id' => $order->get_id(),
				'limit' => 1,
			]
		);

		if ( empty( $expected_note ) ) {
			$this->assertEquals( [], $notes );
			return;
		}
		
		$this->assertCount( 1, $notes );
		$this->assertSame( $expected_note, $notes[0]->content );
```